pkg/dclient/client.go

```diff
@@ -118,7 +118,8 @@ func (c *Client) getResourceInterface(apiVersion string, kind string, namespace
 // Keep this a stateful as the resource list will be based on the kubernetes version we connect to
 func (c *Client) getGroupVersionMapper(apiVersion string, kind string) schema.GroupVersionResource {
 	if apiVersion == "" {
-		return c.DiscoveryClient.GetGVRFromKind(kind)
+		gvr, _ := c.DiscoveryClient.GetGVRFromKind(kind)
```

Comment:
do we need to log the error?

Reply:
What happens if gvr is nil?

Reply:
The error is logged in `GetGVRFromKind`. It can just be an empty GVR object rather than nil.

This change is added particularly to ignore the error when API server is unavailable: https://github.com/realshuting/kyverno/blob/512efcfcde8d0d03a98735c3765fd04856f65261/pkg/utils/util.go#L65-L70